### ISSUE TITLE
Error dialog upon launching application, and application remains empty

### ISSUE THREAD

**Issue by DavidHaslam** [JohnAustinDev/xulsword#242]
I've not run the app for several months, so there've been some Windows Updates meanwhile.

Today, when I launched the app, the flash window did not close automatically and the following error dialog appeared for the first time in my experience.

![screenshot 2017-01-25 12 08 37](https://cloud.githubusercontent.com/assets/16325414/22290189/65994652-e2f7-11e6-83f0-37994b1a3c82.png)

I'm using **Windows 7 x64 Home Premium Edition**, fully up to date.

I have **MK 3.12** installed (2016-02-19).

_I also have **Xiphos** installed and recently installed the **JSword** based **STEP Bible** from Tyndale House._

After I clicked OK, the app itself appeared to be empty !!!

![screenshot 2017-01-25 12 14 05](https://cloud.githubusercontent.com/assets/16325414/22290304/eecd44aa-e2f7-11e6-9153-459074766d42.png)

and the **Help About | More** was also blank.

![screenshot 2017-01-25 12 15 59](https://cloud.githubusercontent.com/assets/16325414/22290330/1359e8fa-e2f8-11e6-80c9-5283212d184e.png)

In the TaskBar there's a thumbnail for a **xulsword-temporary-hidden-window**.

Didn't we see this a long time ago under Windows XP ?

**Comment by DavidHaslam** [comment]
As a check, I'm now installing **MK 3.13** using the file **MK_Setup(WINNT_x86_64-msvc)-3.13.exe** that I dowloaded on 2016-03-21.

**Comment by DavidHaslam** [comment]
I just downloaded and installed **MK 3.14** using the file **MK_Setup(WINNT_x86-msvc)-3.14.exe**

The same thing occurred, except that the dialog appeared twice, and no longer gives any details.

![screenshot 2017-01-25 12 46 55](https://cloud.githubusercontent.com/assets/16325414/22291108/93376f30-e2fc-11e6-8184-5777b43e3210.png)

The app has become unusable on my PC.

**Comment by JohnAustinDev** [comment]
Maybe this is problem caused by an anti virus program, or file permissions? Uninstall MK. Then manually remove any remaining MK files from your user profile by searching there for the folders /IBT/MK and entirely delete them all. Then reinstall MK 3.14 and try again. If there is an unhandled exception (error) it should ask you if you want to report it, and you should say yes.

**Comment by DavidHaslam** [comment]
File permissions shouldn't have changed.

I'm still using **Microsoft Security Essentials** as the AV program.

I also have **MalWareBytes** installed, but not the Premium version.
I could eaisly disable this in Startup using **Piriform CCleaner**.

C:\Users\David\AppData\Roaming\IBT\MK\Profiles
contains five profiles and one **resources** folder.

I have **66 modules** installed in the latter. 
Is it safe to try your instructions without removing these **resources**?

**Comment by DavidHaslam** [comment]
I first stopped **MalWareBytes**.
Then I deleted the five profiles (see above), but leaving the **resources** folder in place.
Then I uninstalled MK successfully.
Then I installed MK 3.14 again. It created a new profile, just as you described.
After installation finished, and MK was opened, I got the same error message.

![screenshot 2017-01-29 17 34 50](https://cloud.githubusercontent.com/assets/16325414/22406348/b5911b44-e649-11e6-86d6-9e0f28f7ad2f.png)

It did NOT ask me if I wanted to report it!

_Please advise further_.

**Comment by DavidHaslam** [comment]
The path 
C:\Users\David\AppData\Roaming\IBT\MK\Profiles\mwd1g468.default\datareporting
contains a file called **aborted-session-ping**

I have zipped it in order to upload it here. 
_It may contains some clues, but it's not something I understand._

[aborted-session-ping.zip](https://github.com/JohnAustinDev/xulsword/files/737684/aborted-session-ping.zip)

**Comment by JohnAustinDev** [comment]
I haven't seen this problem before, and the aborted-session-ping suggest a hard crash of a Mozilla module. My best guess at this point is still interference from anti-virus or something. Security programs are constantly updating and changing, so it could be due to a changed security program, or operating system permission or other setting, rather than xulsword. (?)

You were fine to leave the resources directory and delete the rest. But you can try moving the resources directory somewhere else to save it, and then wiping out Roaming/IBT/MK entirely. There are also quickstart files stored by xulrunner that might be hanging around. So search for all IBT/MK directories below your /users/<name> (even hidden ones), and delete them as well, before reinstalling MK.

If you have a ConsoleLog.txt file in your MK/profile directory please upload that as it would be most helpful (if there is one- but the crash may prevent its writing).

**Comment by DavidHaslam** [comment]
This is what I did today:
- Uninstalled MK 3.14
- Removed the one profile
- Moved the resources folder to somewhere on drive D
- Deleted C:\Users\David\AppData\Roaming\IBT
- Deleted C:\Users\David\AppData\Local\IBT
- Quit MalWareBytes (free) just in case.
- Installed MK 3.14 but unticked Run in final screen.
- Examined C:\Users\David\AppData\Roaming\IBT\MK\Profiles\resources\mods.d
This contains .conf files for the eight modules installed by default with MK

Not yet run the program. Will report on that in next comment.

**Comment by DavidHaslam** [comment]
After running MK, the same error message appeared again (twice).
In the TaskBar the thumbnail for a **xulsword-temporary-hidden-window** appeared transiently.
A profile had been created:
`C:\Users\David\AppData\Roaming\IBT\MK\Profiles\hz8afaat.default`
The main program opened but appeared empty with three windows. See screenshot.

![screenshot 2017-02-02 10 17 20](https://cloud.githubusercontent.com/assets/16325414/22545573/2c3f1ac2-e931-11e6-95a1-ecf0ae9d81eb.png)

Selecting **Windows | 1 Window** did nothing.
No resources were available to display.

The file `C:\Users\David\AppData\Roaming\IBT\MK\Profiles\hz8afaat.default\consoleLog.txt`
was empty (zero bytes).

**Comment by DavidHaslam** [comment]
Ah! But upon closing the program, the same file was updated. Hereby attached.

[consoleLog.txt](https://github.com/JohnAustinDev/xulsword/files/747428/consoleLog.txt)

The 17 modules this records as dropping have nothing to do with MK.
These are actually in my default Sword path, as used by **Xiphos**.

`C:\Users\David\AppData\Roaming\Sword`

**Comment by DavidHaslam** [comment]
Observing the first reported error in consoleLog.txt - I just looked in
`C:\Program Files (x86)\IBT\MK\xulrunner`
There was no such file as **chrome.manifest**

**Comment by DavidHaslam** [comment]
Observing other contents of the file, I couldn't help wondering why MK needs to do anything with **Telemetry**.

Line 59 reads:

`[JavaScript Error: "1486030496132	Toolkit.Telemetry	ERROR	TelemetryStorage::loadAbortedSessionPing - error removing ping: PingReadError JS Stack trace: PingReadError@TelemetryStorage.jsm:80:15 < TelemetryStorageImpl.loadPingFile<@TelemetryStorage.jsm:1440:13" {file: "resource://gre/modules/Log.jsm" line: 749}]`

**Comment by JohnAustinDev** [comment]
Thank you for including ConsoleLog.txt, that was very helpful. MK does look in your default Sword path for modules (just like Xiphos does). One or more of these modules is causing the problem. I see there is a module named: "ACF 2007". MK does not expect modules with spaces in their module code name for instance. When did this become allowable? This could be the culprit...

**Comment by DavidHaslam** [comment]
**ACF2007** is not one from CrossWire or any of the other repositories listed in our MRL.

That Portuguese Bible module was recently (2017-01-19) downloaded from [Siriarah](https://siriarah.wordpress.com/downloads/).

I have just edited my copy of the conf file to remove the space.
I also left a comment in his comments page that the conf file requires correcting.

**Xiphos** had no difficulties displaying it. It's not that a space has become allowed.
Rather perhaps it's that some front-ends are more forgiving than others.

Even so, this wasn't the first error listed in `ConsoleLog.txt`, was it?
It can't account for the missing `chrome.manifest`, can it?

I will try opening MK again now that I've made the correction.

**Comment by JohnAustinDev** [comment]
Much of what is reported in ConsoleLog.txt is inconsequential, including the chrome.manifest error (Mozilla code looks for this file in many places and eventually finds it).

Typically, look for "JavaScript Error"s reported in chrome://xulsword files

**Comment by DavidHaslam** [comment]
Not yet tried to open MK again.

First I needed to tweak five modules from STEP Bible for which [ModName] contained a hyphen.
As only valid characters are the PCRE class [A-Aa-z0-9_] this was a sensible precaution.
I have also advised the STEP development team leader accordingly.

Will try again later today.

**Comment by DavidHaslam** [comment]
Tried MK again after those tweaks to various modules.

Got further this time. The **xulsword-temporary-hidden-window** appeared transiently again.


The following error message also appeared twice when I opened the program.

![screenshot 2017-02-03 15 56 20](https://cloud.githubusercontent.com/assets/16325414/22598048/ff2bc50e-ea29-11e6-8b9c-361db72591f3.png)

However, the main app window appeared, with the first several module tabs.

![screenshot 2017-02-03 16 17 57](https://cloud.githubusercontent.com/assets/16325414/22598751/7d35af4e-ea2c-11e6-9842-68a3940272ad.png)

FIO: Module **ABU** = American Bible Union New Testament (hence blank for Gen.1)

I tried a few tab clicks and eventually got one displayed.

![screenshot 2017-02-03 16 20 07](https://cloud.githubusercontent.com/assets/16325414/22598851/c0d404b2-ea2c-11e6-9587-3b7ca87790e9.png)

But now observe the repeated set of tabs after the half way point!

Unable to consistently select another module after I reached this point.
Sometimes it works, sometimes not.
Some controls work but not correctly. I just ended up with this weird window layout.

![screenshot 2017-02-03 16 23 54](https://cloud.githubusercontent.com/assets/16325414/22599031/4920399e-ea2d-11e6-8ee3-fbac1e23721c.png)

After closing the program, here's the new consoleLog.txt file:

[consoleLog.txt](https://github.com/JohnAustinDev/xulsword/files/751003/consoleLog.txt)

Given that at least some progress has been made by making corrections to installed modules, I am now wondering which other modules might be causing all this remaining grief.

**Comment by DavidHaslam** [comment]
Any advice before my next try would be helpful.

**Comment by JohnAustinDev** [comment]
So it does appear to be the experimental or non-standard modules which are causing the problems. I would like to make MK handle these instances more gracefully. I would suggest saving all your C:\Users\David\AppData\Roaming\Sword modules somewhere else, and then deleting those modules. Then install particular modules one at a time so you know which one(s) cause problems, and report which modules those are. I will probably need access to those modules to insure all issues are dealt with. Are the offending modules available in a standard SWORD repository? If so, this is much higher priority.

**Comment by DavidHaslam** [comment]
Whenever I install a new module or make one myself, it's always in the normal Sword path.

The xulsword modules that I had earlier times installed from IBT were in the resources folder, the one that I moved to somewhere innocuous two days ago. That was not moved back yesterday. 

Rather than moving all the modules, it would be simpler to temporarily move merely the most recent conf files away from the mods.d folder. This could be done in chunks, using the successive approximation method to home in to the largest set of modules that allows MK to work without errors.

Front-ends don't care about extra folders in the ztext folder if there's no configured DataPath pointing to them. 

I think we may only have to consider modules made or installed since a year ago.

The possibility remains that eliminating some modules doesn't fix the error. But we shall see.

**Comment by DavidHaslam** [comment]
OK - further progress. I found on Saturday that my tweaks to some of the invalid modules from **Tyndale STEP** were incomplete, in that I'd not renamed the module folders in ztext. I fixed those yesterday.

I just found time now to open **MK 3.14** again. 

**This time there were no error messages.** 

![screenshot 2017-02-07 16 52 48](https://cloud.githubusercontent.com/assets/16325414/22701593/f0e17cda-ed55-11e6-9997-6cd38006aad7.png)

All is not yet well, but I should now close this issue and create a new issue for the fresh symptoms.
